core/enclave/src/bridges/ecalls.rs

```diff
@@ -65,35 +90,63 @@ pub unsafe extern "C" fn ecall_get_state(
     challenge: &RawChallenge, // 32 bytes randomness for avoiding replay attacks.
     mem_id: u32,
     state: &mut EnclaveState,
-) -> sgx_status_t {
-    let sig = Signature::from_bytes(&sig[..])
-        .expect("Failed to read signatures.");
-    let pubkey = PublicKey::from_bytes(&pubkey[..])
-        .expect("Failed to read public key.");
-    let key = UserAddress::from_sig(&challenge[..], &sig, &pubkey)
-        .expect("Failed to generate user address.");
+) -> EnclaveStatus {
+    let sig = match Signature::from_bytes(&sig[..]) {
+        Ok(sig) => sig,
+        Err(_) => {
+            debug!("Failed to read signatures.");
+            return EnclaveStatus(1);
```

Comment:
Define `EnclaveStatus::error()` cause `EnclaveStatus(1)` is not easy to understand.